src/main/java/spleetwaise/transaction/model/transaction/TransactionIdUtil.java

```diff
@@ -0,0 +1,27 @@
+package spleetwaise.transaction.model.transaction;
+
+import java.util.UUID;
+
+/**
+ * Utilities class for generating transaction UUIDs.
+ */
+public class TransactionIdUtil {
```

Comment:
it may be useful to use structure similar to [this](https://stackoverflow.com/a/51802785), so that there's a clearer separation between code for testing and legitimate util code.

Reply:
so like in the production code, it uses `UuidProvider` which is an interface with `uuid()` method. 
Inject `UuidProvider.Fake` in your tests, and `RandomUuidProvider` in your production code.